File: editor/en/[950]Reveal Cards In Increasing Order.java

//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public int[] deckRevealedIncreasing(int[] deck) {
        Arrays.sort(deck);
        Deque<Integer> queue = new ArrayDeque<>();

        for (int i = deck.length - 1; i >= 0; i--) {
            if (!queue.isEmpty()) {
                queue.add(queue.poll());
            }

            queue.add(deck[i]);
        }

        int[] result = new int[deck.length];

        for (int i = deck.length - 1; i >= 0; i--) {
            result[i] = queue.poll();
        }

        return result;
    }
}
//leetcode submit region end(Prohibit modification and deletion)

File: editor/en/[950]Reveal Cards In Increasing Order.py
# leetcode submit region begin(Prohibit modification and deletion)
from collections import deque


class Solution:
    def deckRevealedIncreasing(self, deck: List[int]) -> List[int]:
        deck.sort()
        queue = deque()

        for card in reversed(deck):
            if queue:
                queue.appendleft(queue.pop())
            queue.appendleft(card)

        return list(queue)

# leetcode submit region end(Prohibit modification and deletion)
